File: game/src/Game.cpp
// #include "Game.h"
// #include <iostream>

// namespace GoudEngine {

// Game::Game() : engine(std::make_unique<Engine>()) {}

// Game::~Game() {
//     Shutdown();
// }

// void Game::SetWindowTitle(const std::string& title) {
//     windowTitle = title;
// }

// void Game::SetWindowSize(int width, int height) {
//     windowWidth = width;
//     windowHeight = height;
// }

// bool Game::Initialize() {
//     if (!engine->Initialize()) {
//         std::cerr << "Engine initialization failed." << std::endl;
//         return false;
//     }
//     std::cout << "Game initialized with window title: " << windowTitle 
//               << " and size: " << windowWidth << "x" << windowHeight << std::endl;
//     return true;
// }

// void Game::Run() {
//     engine->Run();
// }

// void Game::Shutdown() {
//     engine->Shutdown();
//     std::cout << "Game shutting down." << std::endl;
// }

// } // namespace GoudEngine

File: src/game.rs
use crate::engine::Engine;

pub struct Game {
    engine: Engine,
}

impl Game {
    pub fn new() -> Self {
        Self {
            engine: Engine::new(),
        }
    }

    pub fn start(&mut self) {
        // Initialize game-specific elements here
        println!("Starting game...");
        self.engine.run();
    }
}
